docs/launcher.md

```diff
@@ -1,106 +1,628 @@
+
+```json
+{
+  "total_instances": 3,
+  "running_instances": 2,
+  "instances": [
+    {
+      "status": "running",
+      "instance_id": "id-1",
+      "pid": 12345
+    },
+    {
+      "status": "stopped",
+      "instance_id": "id-2",
+      "pid": 12346
+    },
+    {
+      "status": "running",
+      "instance_id": "id-3",
+      "pid": 12347
+    }
+  ]
+}
+```
+
+**Possible Status Values:**
+
+- `running`: Instance is currently running
+- `stopped`: Instance process has stopped
+
+---
+
+#### Get Single Instance Status
+
+**GET** `/v2/vllm/instances/{instance_id}`
+
+Get status information for a specific instance.
+
+**Path Parameters:**
+
+- `instance_id`: ID of the instance
+
+**Response (200 OK):**
+
+```json
+{
+  "status": "running",
+  "instance_id": "instance-id",
+  "pid": 12345
+}
+```
+
+**Possible Status Values:**
+
+- `running`: Instance is currently running
+- `stopped`: Instance process has stopped
+
+**Error Responses:**
+
+- `404 Not Found`: Instance not found
+
+## Usage Examples
+
+### Example 1: Basic Single Instance
 
 ```bash
-curl -X GET \
+# Create instance
+curl -X POST http://localhost:8001/v2/vllm/instances \
   -H "Content-Type: application/json" \
-  http://localhost:8000/v1/vllm
+  -d '{
+    "options": "--model facebook/opt-125m --port 8000"
+  }'
+
+# Use the instance (vLLM API)
+curl http://localhost:8000/v2/models
+
+# Delete instance
+curl -X DELETE http://localhost:8001/v2/vllm/instances/abc123...
 ```
 
-As a result, a JSON reply (like in the example below) will be sent to the client:
+### Example 2: Multiple Models on Different Ports
 
-```json
+```bash
+# Start Llama 2 on port 8010
+curl -X PUT http://localhost:8001/v2/vllm/instances/llama2 \
+  -H "Content-Type: application/json" \
+  -d '{
+    "options": "--model meta-llama/Llama-2-7b-hf --port 8010"
+  }'
+
+# Start GPT-2 on port 8011
+curl -X PUT http://localhost:8001/v2/vllm/instances/gpt2 \
+  -H "Content-Type: application/json" \
+  -d '{
+    "options": "--model gpt2 --port 8011"
+  }'
+
+# Start OPT on port 8012
+curl -X PUT http://localhost:8001/v2/vllm/instances/opt \
+  -H "Content-Type: application/json" \
+  -d '{
+    "options": "--model facebook/opt-1.3b --port 8012"
+  }'
+
+# List all instances
+curl http://localhost:8001/v2/vllm/instances
+```
+
+### Example 3: Using Environment Variables
+
+```bash
+curl -X POST http://localhost:8001/v2/vllm/instances \
+  -H "Content-Type: application/json" \
+  -d '{
+    "options": "--model meta-llama/Llama-2-7b-hf --port 8000 --tensor-parallel-size 2",
+    "env_vars": {
+      "CUDA_VISIBLE_DEVICES": "0,1",
+      "VLLM_ATTENTION_BACKEND": "FLASHINFER",
+      "MAX_BATCH_SIZE": "128"
+    }
+  }'
+```
+
+### Example 4: List Status of All Instances
+
+```bash
+# Get detailed status
+curl http://localhost:8001/v2/vllm/instances
+```
+
+## Configuration
+
+### vLLM Options
+
+The `options` field contains stuff added to the command line of the launched `vllm serve`. Common options include:
```

Comment:
This paragraph ends with an incomplete sentence. (Maybe you think that is not where the paragraph ends. But Markdown is clear on this matter.)